### PR TITLE
log-backup: support the ddl(exchange partition) when PiTR 

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/34742

Problem Summary:
PiTR can not restore the DDL about `exchange partition`.

### What is changed and how it works?
PiTR can support the DDL `exchange partiiton` when rewrite tableInfo

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

### Manual test
The test report
https://pingcap.feishu.cn/docx/doxcnNvs81Is4oYmaBZ0rcAdqmf

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Leavrth — APPROVED** [review]

File: br/pkg/stream/rewrite_meta_rawkv.go [hunk 1]
```diff
@@ -261,10 +287,15 @@ func (sr *SchemasReplace) rewriteTableInfo(value []byte, dbID int64) ([]byte, bo
 
 	tableReplace, exist := dbReplace.TableMap[tableInfo.ID]
 	if !exist {
-		newID, err := sr.genGenGlobalID(context.Background())
-		if err != nil {
-			return nil, false, errors.Trace(err)
+		newID, exist = sr.globalTableIdMap[tableInfo.ID]
+		if !exist {
+			newID, err = sr.genGenGlobalID(context.Background())
```
Comment (YuJuncen) [inline br/pkg/stream/rewrite_meta_rawkv.go:292]: If we don't want to adapt context for now, I think `context.TODO()` would be better.
```suggestion
			newID, err = sr.genGenGlobalID(context.TODO())
```

**Review by YuJuncen — APPROVED** [review]
rest lgtm

**Pull request merged**